src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -53,135 +53,159 @@ namespace impl {
  * If Serialize=false, this function simply calls Kokkos::parallel_reduce().
  * If result contains a value, the output is result+reduction
  * For typical application, begin and end are pack indices.
- * NOTE: we do not provide an overload with Serialized defaulted, since this fcn is an impl
- *       detail, and, normally, should not be used by customer apps
+ * WARNING! This fcn should *not* be used outside this file. The implementation *relies*
```

Comment:
This is fine. Technically, we could return `result` to enforce this comment, and probably the compiler would use return value optimization to make that free, but we can't be sure. See below for further comments concerning the issue of whether `results`'s input value is meaningful.